fun/counting.go.tmpl

```diff
@@ -0,0 +1,110 @@
+      {{ $wrong := "not the right number bud 😅" }}
+    {{/* Message when they try to count twice in a row */}}
+      {{ $twice := "we do counting together 😠" }}
+    {{/* Milestone notification for every 1000 */}}
+      {{ $1000_ntf := "Holy moly, have we really reached" }}
+    {{/* Milestone notification for every 100 */}}
+      {{ $100_ntf := "We have reached" }}
+
+  {{/* Warnings */}}
+    {{/* Info message when the latest correct count was changed (edited or deleted) */}}
+      {{ $change_msg := "we're currently at:" }}
+    {{/* Warn when a member deletes their latest correct count */}}
+      {{ $warn_on_del := true }}
+    {{/* Warning message */}}
+      {{ $warn_msg := "Don't change your latest correct count!" }}
+    {{/* Role to ignore when trying to warn. This should be the roles above the bot since it can't warn members with a role higher than itself. */}}
+      {{ $staff_roles := cslice 763447831829938176 764103587223044116 778952687986802698 764054381535821825 775003755842109440 }}
+
+
+{{/* CODE - Don't edit this part */}}
+{{ $c := sdict }}{{ with (dbGet 20 "counting").Value }}{{ $c = sdict . }}{{ end }}
+{{ $e := execAdmin "prefix" }}{{ $p := reReplace `[*\\_\~]` (slice $e (add 15 (len (str .Guild.ID))) (sub (len $e) 1)) `\$0` }}
+{{ $r := 0 }}{{ with (dbGet .User.ID "resetcount").Value }}{{ $r = . }}{{ end }}
+{{ $s := false }}{{ range .Member.Roles }}{{ if in $staff_roles . }}{{ $s = true }}{{ end }}{{ end }}
+{{ if not $c }}
+  {{ addReactions "☑" }}
+  {{ dbSet 20 "counting" (sdict "l" (sdict "c" 1 "m" 0 "u" 0 "w" 0)) }}
+  {{ print .User.Mention ", <#" .Channel.ID "> is all set up as your counting channel!" }}
+{{ else if and $s (eq (print $p "resetcount") .Message.Content) }}
+  {{ deleteTrigger 1 }}
+  {{ if $s }}
+    {{ if not $r }}
+      {{ $m := sendMessageRetID nil (cembed
+        "color" 16734296
+        "description" (print "‼ " .User.Mention ", are you sure you want to reset the count to **1**?\n\nIf you're sure, type `-resetcount` again.")
+        "footer" (sdict "text" "This message will self destruct in 30 seconds")) }}
+      {{ deleteMessage nil $m 30 }}
+      {{ dbSetExpire .User.ID "resetcount" (sdict "s" 1 "m" $m) 30 }}
+    {{ else }}
+      {{ deleteMessage nil $r.m 0 }}
+      {{ dbDel .User.ID "resetcount" }}
+      {{ range $k, $v := $c }}
+        {{ if eq $k "l" }}{{ $c.Set $k (sdict "c" 1 "m" 0 "u" 0 "w" 0) }}{{ end }}
+      {{ end }}
+      {{ dbSet 20 "counting" $c }}
+      {{ print .User.Mention ", reset the count in <#" .Channel.ID ">!" }}
+      {{ deleteResponse 10 }}
+    {{ end }}
+  {{ else }}
+    {{ print .User.Mention ", you don't have the permissions to reset the count!" }}
+    {{ deleteResponse 5 }}
+  {{ end }}
+{{ else if eq .User.ID $c.l.u }}
+  {{ deleteTrigger 1 }}
+  {{ print .User.Mention ", " $twice }}
+  {{ deleteResponse 5 }}
+{{ else if or (not (reFind `\A\d+\z` .Message.Content)) (ne $c.l.c (toInt .Message.Content)) }}
```

Comment:
Dunno if this is a good idea but you could check for leading zeros here instead of lower down in the chain - regex `[1-9]\d*` would work fine for preventing leading zeros.

Reply:
The error message wouldn't really make sense if I do it there. Having a specific error message to why their count was not accepted is better in my opinion. So I'm going to leave it as it is.